src/signalrclient/web_request.cpp

```diff
@@ -7,8 +7,9 @@
 
 namespace signalr
 {
-    web_request::web_request(const web::uri &url)
+    web_request::web_request(const web::uri &url, web::http::client::http_client_config client_config)

```

Comment:
This is quite inconsistent here. Headers are set using the `set_headers` method. Config is passed to the ctor. And we also have the `set_user_agent` method. 
Again, I think headers should be part of the config. The config could be passed to the ctor or to the `get_response()` but it is really used only in the `get_response()` method so probably it can be passed there without having to store it at all